src/buffer.rs

```diff
@@ -14,7 +14,7 @@ use crate::{
 };
 
 /// A buffer cell
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
```

Comment:
Nit: can we make the order of the derives the same. This is mostly because I'm a little OCD about things that are the same should look the same, but it does have the benefit of being really easy to see at a glance which things are supported if the traits are always in the same order. Perhaps use the following order: Debug, Default, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash.

Taking a look at Ord/PartialOrd and the things which implement these, perhaps it doesn't make sense for most of the types to implement this. Perhaps let's defer that?

Reply:
I agree~